### PR TITLE
feat: uses sandbox acces token when in dev environment

### PR DESCRIPTION
**Pull request by welmaris**
### Issue number

DATA-2017

### Description of fix
When running the script in one of the dev environments, it will use the sandbox access token.
This is based on whether dev shows.

### Other info

- I could only test it for dev1, so it would need to be checked in dev2 but I don't have access to those.
- I also added my local environment to gitignore. This is the standard file created by pyenv virtualenv

### CODE REVIEW

File: wherescape/wherescape.py [hunk 1]
```diff
@@ -71,6 +71,9 @@ def __init__(self):
         self.source_user = os.getenv("WSL_SRCCFG_USER")
         self.source_apikey = os.getenv("WSL_SRCCFG_APIKEY")
 
+        # # source environment
+        # self.source_environment = os.getenv("WSL_SRC_DSN")
+
```
Comment (BFJonk) [inline wherescape/wherescape.py]: 1. not sure why commented code is needed in de code base
2. not sure if WSL_SRC_DSN is actually containing the source_environment

File: wherescape/connectors/hubspot/process_data.py [hunk 2]
```diff
@@ -116,15 +116,14 @@ def compare_names(source_names: list, destination_names: list):
     known_destination_names = []
```
Comment (BFJonk) [inline wherescape/connectors/hubspot/process_data.py:116]: typo in line 109

File: wherescape/connectors/hubspot/collect_data.py [hunk 3]
```diff
@@ -23,16 +24,22 @@ def hubspot_load_data():
...
         hubspot_process_results(access_token, result, column_names, table_name)
         logging.info("hubspot update done")
 
 
-def hubspot_get_token(wherescape_instance: WhereScape, table_name: str):
+def hubspot_get_token(
+    wherescape_instance: WhereScape, table_name: str, develop_env: bool
+):
     """
     Function to get the hubspot access token form the table
```
Comment (BFJonk) [inline wherescape/connectors/hubspot/collect_data.py]: typo form --> from

**Review by BFJonk — COMMENTED** [review]
I think it works, also for dev1. Wondering if other potential users will have a similar naming convention, wouldn't it be nicer or possible to add a parameter to hubspot_load_data(), or maybe even set a Hubpot() object attribute?

**Comment by welmaris** [comment]
> I think it works, also for dev1. Wondering if other potential users will have a similar naming convention, wouldn't it be nicer or possible to add a parameter to hubspot_load_data(), or maybe even set a Hubpot() object attribute?

I'm almost certain not everyone will have the same naming convention as us, but it's specified in the readme for them so as long as they read that it should be fine. 

Though it might be an option to allow personal choice using an optional parameter in hubspot_load_data(). This would also require some changes to hubspot_get_token() so I'd rather do that as its own ticket. In regards to a Hubspot() object attribute, the information is only needed to get the access_token, which is already an attribute, so I feel like it would be excessive / unnecessary. 

I'd prefer not to use a parameter to specify the environment because it seems unnecessary. I can get the needed info from Wherescape itself directly which removes risk of human error.

File: wherescape/connectors/hubspot/collect_data.py [hunk 4]
```diff
@@ -47,6 +54,9 @@ def hubspot_get_token(wherescape_instance: WhereScape, table_name: str):
     table_words = table_name.split("_")
 
     logging.info("retrieving access_token")
+    if develop_env:
+        logging.info("using developmental environment")
+        return parameter_name + "_sandbox"
```
Comment (BFJonk) [inline wherescape/connectors/hubspot/collect_data.py]: Does this work? was this tested? function is called `hubspot_get_token()`, however you're returning just a name not the token.

**Comment by BFJonk** [comment]
I just read the [readme.md](https://github.com/wearespindle/python-wherescape/blob/main/wherescape/connectors/hubspot/readme.md). Missed that.  The section on the environments in the .md  needs to checked on fluency and clarity: `this script will distinguish which To distinguish` Can you please clarify this text as part of the change?

**Comment by BFJonk** [comment]
NB. I would find it more logical that for each environment (voys/spindle), one could have a test env. So `hubspot_access_token_voys_dev` and `hubspot_access_token_spinlde_dev`

**Comment by BFJonk** [comment]
NB2. code looks more logical to me now than before!
NB3.  by squashing your commits I can no longer see what the old code looked like. In the review process, this is a bit annoying.

**Comment by welmaris** [comment]
NB: yeah,I should've considered it and can change it a bit. I did add it now as a possibility so it is an option
NB2: thanks!
NB3: took that habit from Backend where it is required. But where gitlab still keeps it relatively easy to check such changes even with a squash where github  doesn´t so I will avoid doing it here.

**Review by BFJonk — APPROVED** [review]
TRhanks for the changes

**Pull request merged**